### ISSUE TITLE
Support Wolfram Engine

### ISSUE THREAD

**Issue by szhorvat** [mmatera/iwolfram#11]
It would be nice if this worked with http://wolfram.com/engine

Personally, I couldn't yet get it working with standard Mathematica either. The first problem was a space in the path (I have `/Applications/Mathematica 12.0.app` on OS X). Note that the Wolfram Engine installs to such a path by default on Mac. Once I removed the space, and provided the location as 

```
python setup.py install --mma-exec /Applications/Mathematica12.0.app/Contents/MacOS/MathKernel
```

it just hangs, not sure way.

Back to the Wolfram Engine: note that the startup text is different than for Mathematica.

**Comment by Ludwiggle** [comment]
@szhorvat yes you could delete the startup text mathing from the installation script and form the kernel too

**Comment by szhorvat** [comment]
I don't think that's why it hangs because I was trying to get it working with actual Mathematica ... Did you get it working on Mac with M12.0 or Linux only?

**Comment by Ludwiggle** [comment]
I used it only with 11.3 on Linux. I am going to give it a try with the new Wolfram Engine (12) on Linux. 
In the meantime you could have a look at the initialization script called `wmath` inside the kernel folder. That is supposed to run through wolframscript.

**Comment by rljacobson** [comment]
I knew one of you would have beaten me to the punch with this question and its diagnosis. I just want to post a link-back to the SE question about this: https://mathematica.stackexchange.com/a/199040/27662.

**Comment by mmatera** [comment]
Indeed, the current project would require some extra work to be  "ironed"
properly. Sadly, I can work on it just during my scarce extra time.

Dr. Juan Mauricio Matera

CONICET
IFLP - Dto. de Física UNLP,
Casilla de Correos 67
Calle 49 y 115 s/n,
1900 La Plata, Argentina
Tel +54 221 4246062 (int 294)
Fax +54 221 4236335
e-Mail: matera@fisica.unlp.edu.ar
homepage: http://www.fisica.unlp.edu.ar/Members/matera
Skype: mauricio5831

**Comment by rljacobson** [comment]
I completely understand, @mmatera. I am just grateful you have shared your code with the world. :)

**Comment by Ludwiggle** [comment]
If you are looking for something that just works give a try to https://github.com/Ludwiggle/JWLS
If your wolframscript works, JWLS works too, cause JWLS is a bash kernel that reads and writes to fifos; as such, it hardly fails.
